shared/app/nav.native.js

```diff
@@ -109,45 +114,20 @@ const barStyle = ({showStatusBarDarkContent, underStatusBar}) => {
 
 function renderStackRoute(route) {
   const {underStatusBar, hideStatusBar, showStatusBarDarkContent} = route.tags
-  // Skip extra view if no statusbar
-  if (hideStatusBar) {
-    return route.component
-  }
 
   return (
     <Box style={route.tags.underStatusBar ? sceneWrapStyleUnder : sceneWrapStyleOver}>
       <StatusBar
         hidden={hideStatusBar}
         translucent={true}
         backgroundColor="rgba(0, 26, 51, 0.25)"
-        barStyle={barStyle({showStatusBarDarkContent, underStatusBar})}
+        barStyle={barStyle(showStatusBarDarkContent, underStatusBar)}
       />
       {route.component}
     </Box>
   )
 }
 
-const forIOS = ({hideNav, shim, tabBar}) => (
```

Comment:
We now have a single implementation using the KeyboardAvoidingView (with some different params). We listen for keyboard showing/hiding and hide our tabBar ourselves.